scripts/powershell/Verify-And-Merge-PRs.ps1

```diff
@@ -0,0 +1,145 @@
+$ClosedPRs = [System.Collections.ArrayList]::new()
+$UnexpectedState = [System.Collections.ArrayList]::new()
+$MergablePRs = [System.Collections.ArrayList]::new()
+$BlockedPRs = [System.Collections.ArrayList]::new()
+
+
+$GitHubAPIUrlBase = "https://api.github.com/repos/"
+$Headers = @{
+    Authorization = "bearer $AuthToken"
+}
+
+# Get PRData from File
+$PRData = Get-Content $PRDataArtifactPath
+foreach ($line in $PRData)
+{
+    $PRDetails = $line.Split(';')
+    $PRDataObj = [PSCustomObject]@{
+        RepoOwner = $PRDetails[0]
+        RepoName = $PRDetails[1]
+        PRNumber = $PRDetails[2]
+    }
+    [void]$PRObjs.Add($PRDataObj)
+}
+
+# Confirm Mergability
+foreach ($obj in $PRObjs)
+{
+    $APIUrl = "${GitHubAPIUrlBase}$($obj.RepoOwner)/$($obj.RepoName)/pulls/$($obj.PRNumber)"
+    try
+    {
+        $response = Invoke-RestMethod -Headers $Headers $APIUrl
+        if ($response.merged)
+        {
+            continue
+        }
+        elseif ($response.state -eq "closed")
+        {
+            [void]$ClosedPRs.Add($obj)
+        }
+        else 
+        {
+            if ($response.mergeable_state -ne "clean")
+            {
+                [void]$BlockedPRs.Add($obj)
+            }
+            elseif ($response.mergeable -and ($response.mergeable_state -eq "clean"))
+            {
+                $obj | Add-Member -MemberType NoteProperty -Name "HeadSHA" -Value $response.head.sha
+                $obj | Add-Member -MemberType NoteProperty -Name "HeadLabel" -Value $response.head.label
+                [void]$MergablePRs.Add($obj)
+            }
+            else 
+            {
+                [void]$UnexpectedState.Add($obj)
+            }
+        }
+    }
+    catch
+    {
+        Write-Error "Invoke-RestMethod ${APIUrl} failed with exception:`n$_"
+        exit 1
+    }
+}
+
+# Verify and Report all Unmergable PRS
+if (($BlockedPRs.Count -gt 0) -or ($ClosedPRs.Count -gt 0) -or ($UnexpectedState.Count -gt 0))
+{
+    if ($BlockedPRs.Count > 0)
+    {
+        Write-Host "The following PRs are Blocked, Please ensure all check are green and PR is approved"
+        foreach ($obj in $BlockedPRs)
+        {
+            Write-Host "`thttps://github.com/$($obj.RepoOwner)/$($obj.RepoName)/pull/$($obj.PRNumber): State: Blocked"
+        }
+    }
+
+    if ($ClosedPRs.Count > 0)
+    {
+        Write-Host "The following PRs are Closed, Please investigate why a Sync PR was closed prematurely."
+        foreach ($obj in $ClosedPRs)
+        {
+            Write-Host "`thttps://github.com/$($obj.RepoOwner)/$($obj.RepoName)/pull/$($obj.PRNumber): State: Closed"
+        }
+    }
+
+    if ($UnexpectedState.Count > 0)
+    {
+        Write-Host "The following PRs are Unexpected state, Run sync pipeline again"
+        foreach ($obj in $UnexpectedState)
+        {
+            Write-Host "`thttps://github.com/$($obj.RepoOwner)/$($obj.RepoName)/pull/$($obj.PRNumber): State: Unexpected"
+        }
+    }
+
```

Comment:
Add LogError "At least one sync PR is not able to be merged please investigate and then retry running this job to auto-merge them again"